### PR TITLE
chore(*): add codspeed for benchmark visualization

### PR DESCRIPTION
**Pull request by juno7803**
related https://github.com/toss/es-toolkit/issues/57

### Background
Recently, contributions to improve the performance of the es-toolkit have been ongoing. Many people have been measuring performance improvements using the experimental bench feature applied to vitest. ([like this](https://github.com/toss/es-toolkit/pull/53#issue-2353572060))

So I have discovered [codspeed](https://docs.codspeed.io/), a tool that further automates this process, and I want to install it. We can easliy install this package because we already have `*.bench.ts` files. [Prisma](https://github.com/prisma/prisma/pull/24554) is already using it, and it is available for [free](https://codspeed.io/pricing) in public open source projects.

image

### Features
CodSpeed visualizes performance metrics, allowing us to see improvements or regressions by comparing with previous benchmarks. We can view these visualizations both in PR comments and on the codSpeed dashboard. (maybe it would be https://codspeed.io/toss/es-toolkit/benchmarks)

###  Changes
When I tested it in the forked repository, and it was difficult to identify which function belonged to lodash and es-toolkit. So, I added delimiters like `lodash/uniq` and `es-toolkit/uniq` in the `*.bench.ts` file.

image

### CODE REVIEW

**Review by manudeli — DISMISSED** [review]
Way better DX. This motivate contributors intuitively with numbers

File: benchmarks/partition.bench.ts [hunk 1]
```diff
@@ -3,11 +3,11 @@ import { partition as partitionToolkit } from 'es-toolkit';
 import { partition as partitionLodash } from 'lodash';
 
 describe('partition', () => {
-  bench('es-toolkit', () => {
+  bench('es-toolkit/', () => {
```
Comment (raon0211) [inline benchmarks/partition.bench.ts:6]: `es-toolkit/partition` ?

Comment (juno7803) [reply benchmarks/partition.bench.ts]: 🦅👀 I fixed it. Thanks! [3020b5d](https://github.com/toss/es-toolkit/pull/75/commits/3020b5d246dd228336b099ccebed61ebe8d9de7b)

**Review by raon0211 — APPROVED** [review]
Thanks!

**Pull request merged**